README.md

```diff
@@ -37,8 +37,8 @@ cask_args appdir: "~/Applications", require_sha: true
 
 # 'brew install'
 brew "imagemagick"
-# 'brew install --with-rmtp', 'brew services restart' on version changes
-brew "denji/nginx/nginx-full", args: ["with-rmtp"], restart_service: :changed
+# 'brew install --with-rmtp', 'brew link --overwrite', 'brew services restart' on version changes
+brew "denji/nginx/nginx-full", link: :overwrite, args: ["with-rmtp"], restart_service: :changed
```

Comment:
Not sure about this DSL; having `true`/`false`/`:overwrite` as valid values is a bit confusing. Thoughts about `link_overwrite: true` instead?

Reply:
Was thinking of `:overwrite` as a simple way to get at the extra forceful version of `true` (that also implies `true`).
Given the inherent danger of overwrite maybe it is better to have a very clear separate option like you suggest. 
Will users be confused about how `link:` `true` / `false` / `nil` interacts with `link_overwrite: true` ?

Reply:
> Was thinking of `:overwrite` as a simple way to get at the extra forceful version of `true` (that also implies `true`).

Ok, makes sense 👍🏻 

> Will users be confused about how `link:` `true` / `false` / `nil` interacts with `link_overwrite: true` ?

Yes, maybe, unless we make it e.g. error if both are passed?

Reply:
I've worked through the interactions a bit further to give more insight on:
> Not sure about this DSL; having true/false/:overwrite as valid values is a bit confusing. 

> Will users be confused about how link: true / false / nil interacts with link_overwrite: true ?

The role of `nil` looks to be particularly important. It probably is almost never actually specified in a Brewfile but it has the most complex role as an 'adopt defaults' choice that results in directly opposite actions for keg_only vs not_keg_only formulas.

Expanding on the two proposed options for extending the Brewfile dsl we have:
1. adding `:overwrite` as an additional state for the `link:` option

| Brewfile                       | Command run                                       |
|--------------------------------|---------------------------------------------------|
| brew "mysql", link: :overwrite | `brew link --overwrite`                             |
| brew "mysql", link: :true      | `brew link`                                         |
| brew "mysql", link: :false     | `brew unlink`                                       |
| brew "mysql", *[link: :nil]*     | Not_keg_only: `brew link`          Keg_only: `brew unlink` |

2. adding a new `link_overwrite:` option

| Brewfile                                                   | Command run             |
|------------------------------------------------------------|-------------------------|
| brew "gnupg", link:true, link_overwrite: true              | `brew link --overwrite`   |
| brew "gnupg", link:false, *[link_overwrite: false/nil]*      | `brew unlink`             |
| brew "gnupg", link:false, link_overwrite: true             | Not a valid combination |
|                                                            |                         |
| *Not_keg_only*                                             |                         |
| brew "gnupg", *[link:nil]*, link_overwrite: true             | `brew link --overwrite`   |
| brew "gnupg", *[link:nil/true]*, *[link_overwrite: false/nil]* | `brew link`               |
|                                                            |                         |
| *Keg_only*                                                 |                         |
| brew "gnupg@2.2", link:true, *[link_overwrite: false/nil]*   | `brew link`               |
| brew "gnupg@2.2", *[link:nil]*, *[link_overwrite: false/nil]*  | `brew unlink`             |
| brew "gnupg@2.2", *[link:nil]*, link_overwrite: true         | Not a valid combination |

Re: option 2 'Not a valid combination':
1.  `brew unlink --overwrite` is not a thing (it is nonsensical)
2. The combinations are therefore contradictory in intent - the `link` portion suggests `brew unlink` but `link_overwrite:true` suggests user expects to `brew link`. Which should take precedence?
3. A bare `brew "<formula>", link_overwrite: true` is probably going to result in user confusion as it won't work for keg_only but will work for not_keg_only. 
4. We would need to choose to error or do nothing. Based on above points i'd say it should be errors.

Reply:
> It probably is almost never actually specified in a Brewfile but it has the most complex role as an 'adopt defaults' choice that results in directly opposite actions for keg_only vs not_keg_only formulas.

Agreed 👍🏻 



> Expanding on the two proposed options for extending the Brewfile dsl we have:

@yelsa Great work spelling these out. Intentionally or not: I'm now convinced that `link: :overwrite` is the best option here. Thanks!